pkg/inject/inject.go

```diff
@@ -0,0 +1,585 @@
+package inject
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/linkerd/linkerd2/controller/gen/config"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	log "github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	k8sMeta "k8s.io/apimachinery/pkg/api/meta"
+	k8sResource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// LocalhostDNSNameOverride allows override of the controlPlaneDNS. This
+	// must be in absolute form for the proxy to special-case it.
+	LocalhostDNSNameOverride = "localhost."
+	// ControlPlanePodName default control plane pod name.
+	ControlPlanePodName = "linkerd-controller"
+	// PodNamespaceEnvVarName is the name of the variable used to pass the pod's namespace.
+	PodNamespaceEnvVarName = "LINKERD2_PROXY_POD_NAMESPACE"
+	// PrometheusImage is the docker image and tag for the Prometheus instance used in the control plane
+	PrometheusImage                 = "prom/prometheus:v2.7.1"
+	prometheusProxyOutboundCapacity = 10000
+	// DefaultDockerRegistry is the default registry to pull the proxy and init container images from
+	DefaultDockerRegistry = "gcr.io/linkerd-io"
+	// DefaultKeepaliveMs is used in the proxy configuration for remote connections
+	DefaultKeepaliveMs = 10000
+)
+
+var injectableKinds = []string{"Deployment", "ReplicationController",
+	"ReplicaSet", "Job", "DaemonSet", "StatefulSet", "Pod"}
```

Comment:
Whoops, one more comment. Lowercase versions of all of these strings already exist in `pkg/k8s/k8s.go`, so I'd prefer to use those if possible. Something like this:

```diff
diff --git a/pkg/inject/inject.go b/pkg/inject/inject.go
index c64a2c47..8955bc31 100644
--- a/pkg/inject/inject.go
+++ b/pkg/inject/inject.go
@@ -33,8 +33,15 @@ const (
 	defaultKeepaliveMs = 10000
 )
 
-var injectableKinds = []string{"Deployment", "ReplicationController",
-	"ReplicaSet", "Job", "DaemonSet", "StatefulSet", "Pod"}
+var injectableKinds = []string{
+	k8s.DaemonSet,
+	k8s.Deployment,
+	k8s.Job,
+	k8s.Pod,
+	k8s.ReplicaSet,
+	k8s.ReplicationController,
+	k8s.StatefulSet,
+}
 
 // objMeta provides a generic struct to parse the names of Kubernetes objects
 type objMeta struct {
@@ -120,7 +127,7 @@ func (conf *ResourceConfig) GetPatch(bytes []byte) ([]byte, []Report, error) {
 	}
 
 	var patch *Patch
-	if conf.meta.Kind == "Pod" {
+	if strings.ToLower(conf.meta.Kind) == k8s.Pod {
 		patch = NewPatchPod()
 	} else {
 		patch = NewPatchDeployment()
@@ -166,7 +173,7 @@ func (conf *ResourceConfig) GetPatch(bytes []byte) ([]byte, []Report, error) {
 // KindInjectable returns true if the resource in conf can be injected with a proxy
 func (conf *ResourceConfig) KindInjectable() bool {
 	for _, kind := range injectableKinds {
-		if conf.meta.Kind == kind {
+		if strings.ToLower(conf.meta.Kind) == kind {
 			return true
 		}
 	}
@@ -174,20 +181,20 @@ func (conf *ResourceConfig) KindInjectable() bool {
 }
 
 func (conf *ResourceConfig) getFreshWorkloadObj() runtime.Object {
-	switch conf.meta.Kind {
-	case "Deployment":
+	switch strings.ToLower(conf.meta.Kind) {
+	case k8s.Deployment:
 		return &v1beta1.Deployment{}
-	case "ReplicationController":
+	case k8s.ReplicationController:
 		return &v1.ReplicationController{}
-	case "ReplicaSet":
+	case k8s.ReplicaSet:
 		return &v1beta1.ReplicaSet{}
-	case "Job":
+	case k8s.Job:
 		return &batchv1.Job{}
-	case "DaemonSet":
+	case k8s.DaemonSet:
 		return &v1beta1.DaemonSet{}
-	case "StatefulSet":
+	case k8s.StatefulSet:
 		return &appsv1.StatefulSet{}
-	case "Pod":
+	case k8s.Pod:
 		return &v1.Pod{}
 	}
 
@@ -235,8 +242,8 @@ func (conf *ResourceConfig) parse(bytes []byte) error {
 		return nil
 	}
 
-	switch conf.meta.Kind {
-	case "Deployment":
+	switch strings.ToLower(conf.meta.Kind) {
+	case k8s.Deployment:
 		v := obj.(*v1beta1.Deployment)
 		if err := yaml.Unmarshal(bytes, v); err != nil {
 			return err
@@ -250,7 +257,7 @@ func (conf *ResourceConfig) parse(bytes []byte) error {
 		conf.podLabels[k8s.ProxyDeploymentLabel] = v.Name
 		conf.complete(&v.Spec.Template)
 
-	case "ReplicationController":
+	case k8s.ReplicationController:
 		v := obj.(*v1.ReplicationController)
 		if err := yaml.Unmarshal(bytes, v); err != nil {
 			return err
@@ -260,7 +267,7 @@ func (conf *ResourceConfig) parse(bytes []byte) error {
 		conf.podLabels[k8s.ProxyReplicationControllerLabel] = v.Name
 		conf.complete(v.Spec.Template)
 
-	case "ReplicaSet":
+	case k8s.ReplicaSet:
 		v := obj.(*v1beta1.ReplicaSet)
 		if err := yaml.Unmarshal(bytes, v); err != nil {
 			return err
@@ -270,7 +277,7 @@ func (conf *ResourceConfig) parse(bytes []byte) error {
 		conf.podLabels[k8s.ProxyReplicaSetLabel] = v.Name
 		conf.complete(&v.Spec.Template)
 
-	case "Job":
+	case k8s.Job:
 		v := obj.(*batchv1.Job)
 		if err := yaml.Unmarshal(bytes, v); err != nil {
 			return err
@@ -280,7 +287,7 @@ func (conf *ResourceConfig) parse(bytes []byte) error {
 		conf.podLabels[k8s.ProxyJobLabel] = v.Name
 		conf.complete(&v.Spec.Template)
 
-	case "DaemonSet":
+	case k8s.DaemonSet:
 		v := obj.(*v1beta1.DaemonSet)
 		if err := yaml.Unmarshal(bytes, v); err != nil {
 			return err
@@ -290,7 +297,7 @@ func (conf *ResourceConfig) parse(bytes []byte) error {
 		conf.podLabels[k8s.ProxyDaemonSetLabel] = v.Name
 		conf.complete(&v.Spec.Template)
 
-	case "StatefulSet":
+	case k8s.StatefulSet:
 		v := obj.(*appsv1.StatefulSet)
 		if err := yaml.Unmarshal(bytes, v); err != nil {
 			return err
@@ -300,7 +307,7 @@ func (conf *ResourceConfig) parse(bytes []byte) error {
 		conf.podLabels[k8s.ProxyStatefulSetLabel] = v.Name
 		conf.complete(&v.Spec.Template)
 
-	case "Pod":
+	case k8s.Pod:
 		v := obj.(*v1.Pod)
 		if err := yaml.Unmarshal(bytes, v); err != nil {
 			return err
```

I went ahead and added that change to #2422.